src/imagej/__init__.py

```diff
@@ -1517,6 +1522,24 @@ def _includes_imagej_legacy(items: list):
     return any(item.startswith("net.imagej:imagej-legacy") for item in items)
 
 
+def _macos_is_main_thread():
+    """Detect if the current thread is the main thread on macOS.
+
+    :return: Boolean indicating if the current thread is the main thread.
+    """
+    # try to load the pthread library
+    try:
+        pthread = cdll.LoadLibrary("libpthread.dylib")
+    except OSError as e:
+        print("No pthread library found.", e)
```

Comment:
Did you test what happens when this is thrown? I'd assume that you'd then get some weird error below in the `if` block below- seems like you should return `False` here?

Reply:
Hmm fair enough 😅. I agree returning False here with a print statement about the missing library is the best solution here.

Reply:
Fixed/improved in 972592dda94f983b51ea3c3e1f9b2ae390ddbde9.